### ISSUE TITLE
User categories do not work

### ISSUE THREAD

**Issue by simonbcn** [GChristensen/torrent-add#6]
Tested in [Arc](https://arc.net/) (_derived from Chromium_) and Google Chrome.
```
Add Torrent To v0.3
```

When I add a torrent using the `User categories`, it sends the torrent but without category. It works fine if I use the `Torrent client categories`.

**Comment by simonbcn** [comment]
I forgot to say that the user categories I have put are the same that exist in the torrent client. 
The only thing is that I use a reduced version of the client categories.

**Comment by simonbcn** [comment]
Separate use case?
I have 8 categories in client and it works well with extension if I use `Torrent client categories`.
But when I use 3 of those categories as `User categories` it does not work.

**Comment by simonbcn** [comment]
I have seen the code and I think the solution is very simple:
```
qbittorrent.js:
....
    async #addTorrent(category, form) {
        try {
            await this.#login();

            const prefs = await this.#fetchJSON("app", "preferences");

            form.append("category", category);

            const apiURL = this.#makeAPIURL("torrents", "add");
            const response = await fetch(apiURL, {method: "POST", body: form});

            if (!response.ok)
                showNotification("Error adding torrent.");
        }
        finally {
            await this.#logout();
        }
    }
....
```

This way qBittorrent will create the category (_and download folder_) if it does not exist or use an existing category/folder.

There is no need for the additional processing you currently do when using `User Categories` in qBittorrent.

**Comment by GChristensen** [comment]
Yes, probably. But initially the client categories were not processed at all. There were only folder mode, probably it was renamed not correctly. I thing, there should be a folder mode, user category mode, and client category mode.

**Comment by simonbcn** [comment]
In qBittorrent there is no folder mode (_in fact, currently the client does not receive any folders when set to `User Categories` mode_), there are only tags/categories.

**Comment by GChristensen** [comment]
In the API there is an ability to set plain path without setting a category. This was the initial mode, and it does not require additional configuration, as the use of categories do. 
I've added the ability to specify user categories in the settings. This will be available in v0.4, when it will be reviewed/approved and available for update in the stores.
